manim/mobject/geometry.py

```diff
@@ -1554,3 +1556,236 @@ def __init__(self, main_shape, *mobjects, **kwargs):
+                    angle_fin = angle_2 - angle_1
+                else:
+                    angle_fin = 2 * np.pi - (angle_1 - angle_2)
+            else:
+                start_angle = angle_1
+                if angle_2 < angle_1:
+                    angle_fin = -angle_1 + angle_2
+                else:
+                    angle_fin = -2 * np.pi + (angle_2 - angle_1)
+
+            Arc.__init__(
+                self,
+                radius=radius,
+                angle=angle_fin,
+                start_angle=start_angle,
+                arc_center=inter,
+                **kwargs
+            )
+            if dot == True:
+                if dot_radius == None:
+                    dot_radius = radius / 10
+                else:
+                    self.dot_radius = dot_radius
+                right_dot = Dot(ORIGIN, radius=dot_radius, color=dot_color)
+                dot_anchor = (
+                    inter
+                    + (self.get_center() - inter)
+                    / np.linalg.norm(self.get_center() - inter)
+                    * radius
+                    * dot_distance
+                )
+                right_dot.move_to(dot_anchor)
+                self.add(right_dot)
+
+
+class RightAngle(Angle):
+    """An elbow-type mobject representing a right angle between two lines.
+
+    Parameters
+    ----------
+    line1 : :class:`Line`
+        The first line.
+    line2 : :class:`Line`
+        The second line.
+    length : :class:`float`
+        The length of the arms.
+    kwargs
+        Further keyword arguments that are passed to the constructor of :class:`~.VMobject`.
+
+    Examples
+    --------
+
+    ..manim:: RightAngleExample
+        :save_last_frame:
+
+        class RightAngleExample(Scene):
+            def construct(self):
+                line1 = Line( LEFT, RIGHT )
+                line2 = Line( DOWN, UP )
+                rightangles = [
+                    RightAngle(line1, line2),
+                    RightAngle(line1, line2, length=0.4, quadrant=(1,-1)),
+                    RightAngle(line1, line2, length=0.5, quadrant=(-1,1), stroke_width=8),
+                    RightAngle(line1, line2, length=0.7, quadrant=(-1,-1), color=RED),
+                ]
+                line_list = VGroup( *[VGroup() for k in range(4)] )
+                for k in range(4):
+                    linea = line1.copy()
+                    lineb = line2.copy()
+                    line_list[k].add( linea )
+                    line_list[k].add( lineb )
+                    line_list[k].add( rightangles[k] )
+                line_list.arrange_in_grid(buff=1.5)
+                self.add(
+                    line_list
+                )
+
+    """
+
+    def __init__(self, line1, line2, length=None, **kwargs):
+        if length == None:
+            Angle.__init__(self, line1, line2, elbow=True, **kwargs)
+        else:
+            Angle.__init__(self, line1, line2, radius=length, elbow=True, **kwargs)
```

Comment:
I think this can be simplified to just:
```suggestion
        Angle.__init__(self, line1, line2, radius=length, elbow=True, **kwargs)
```
since if `length==None` then the Angle constructor would just receive radius=None and calculate it dynamically.